app/scripts/metamask-controller.js

```diff
@@ -164,6 +170,18 @@ module.exports = class MetamaskController extends EventEmitter {
       encryptor: opts.encryptor || undefined,
     })
 
+    // permissions
+    this.initializePermissions()
+
+    // domain login management
+    this.logins = {}
+    if (!('logins' in initState)) {
```

Comment:
you use the `in` pattern a lot but i think it can be dangerous
as `{ logins: undefined }` or `{ logins: null }` will pass this test